src/DensityTest.jl

```diff
@@ -0,0 +1,272 @@
+# Density test for manipulation in the running variable. 
+# This work is based on McCrary, J. (2008). Manipulation of the running variable in the regression discontinuity design: A density test. Journal of econometrics, 142(2), 698-714.
+
+"""
+    McCraryTest(bin::Union{Real,Nothing} = nothing, bw::Union{Real,Nothing} = nothing) 
+
+A McCrary (2008) test for manipulation in the running variable for `SharpRD` estimator.
+
+The `bin` object is the width of each bin. The default is 2*sd(runvar)*length(runvar)^(-0.5), following McCrary 2008.
+
+The `bw` object is the bandwidth. The default uses the calculation from McCrary (2008), pg. 705.
+"""
+Base.@kwdef struct McCraryTest 
+    bin::Union{Real,Nothing} = nothing 
+    bw::Union{Real,Nothing} = nothing 
+end
+
+Base.@kwdef struct FittedMcCraryTest{T,M,V,C}
+    b::T
+    h::T
+    θhat::T
+    σθ::T
+    pval::T
+    Js::M
+    fitted_res::V
+    coeftable::C
+end
+
+
+# 0. Kernel used in the McCrary Test
+
+TriangularKernel(t) = max(0, (1 - abs(t)))
+
+# 1. Descriptive Statistics
+
+function DescriptiveStats(RunVar)
+    n = Base.length(RunVar)
+    sd = StatsBase.std(RunVar)
+    rmax = Base.maximum(RunVar)
+    rmin = Base.minimum(RunVar)
+    return n, sd, rmax, rmin
+end
+
+# 2. Binsize (uses DescriptiveStats)
+# function Binsize(n, sd, bin::Union{Real,Nothing}=nothing)
+function Binsize(n, sd)
+    b = 2 * sd * n^(-.5)
+end
+
+# 3. Number of Binsize
+function BinNumber(c, b, rmin, rmax)
+    Jl = convert(Int64, round(floor((c - rmin) / b) + 1))
+    J = convert(Int64, floor((rmax - rmin) / b) + 2)
+    Jr = J - Jl
+    return J, Jl, Jr
+end
+
+# 4. Discretetized running variable (DRV).
```

Comment:
Would it be possible to use the [`DiscretizedRunningVariable`](https://github.com/nignatiadis/RegressionDiscontinuity.jl/blob/74051523aee41b6ab79d21e4e92770f4b2773556/src/running_variable.jl#L60) type that Evan introduced here?  This would allow to reuse this code and have a good central implementation of discretization that we then use for many other functionalities. I suspect that to get it to work one would need to modify [`fit`](https://github.com/nignatiadis/RegressionDiscontinuity.jl/blob/74051523aee41b6ab79d21e4e92770f4b2773556/src/running_variable.jl#L99) to also take a binwidth argument.

Reply:
I did that! :-) I updated `fit` to also take bin width argument. `DiscretizedRunningVariable` now has two methods: one that takes both number of bins and bin width as arguments, and a second one that only ask for number of bins.